contract_compassion/model/contracts.py

```diff
@@ -0,0 +1,646 @@
+                                             invoice.id, 'invoice_open', cr)
+
+    def _is_fully_managed(self, cr, uid, ids, field_name, arg, context):
+        """Tells if the correspondent and the payer is the same person."""
+        res = dict()
+        for contract in self.browse(cr, uid, ids, context=context):
+            res[contract.id] = contract.partner_id == contract.correspondant_id
+        return res
+
+    def get_ending_reasons(self, cr, uid, context=None):
+        """Returns all the ending reasons of sponsorships"""
+        return [
+            ('1', _("Depart of child")),
+            ('2', _("Mistake from our staff")),
+            ('3', _("Death of sponsor")),
+            ('4', _("Moved to foreign country")),
+            ('5', _("Not satisfied")),
+            ('6', _("Doesn't pay")),
+            ('8', _("Personal reasons")),
+            ('9', _("Never paid")),
+            ('10', _("Subreject")),
+            ('11', _("Exchange of sponsor")),
+            ('12', _("Financial reasons")),
+            ('25', _("Not given")),
+        ]
+
+    def _get_channels(self, cr, uid, context=None):
+        """Returns the available channel through the new sponsor
+        reached Compassion.
+        """
+        return [
+            ('postal', _("By mail")),
+            ('direct', _("Direct")),
+            ('email', _("By e-mail")),
+            ('internet', _("From the website")),
+            ('phone', _("By phone")),
+            ('payment', _("Payment")),
+        ]
+
+    def _has_mandate(self, cr, uid, ids, field_name, args, context=None):
+        # Search for an existing valid mandate
+        res = dict()
+        for contract in self.browse(cr, uid, ids, context):
+            count = self.pool.get('account.banking.mandate').search(cr, uid, [
+                ('partner_id', '=', contract.partner_id.id),
+                ('state', '=', 'valid')], count=True, context=context)
+            res[contract.id] = bool(count)
+        return res
+
+    def _name_get(self, cr, uid, ids, field_name, args, context=None):
+        return {c[0]: c[1] for c in self.name_get(cr, uid, ids, context)}
+
+    ###########################
+    #        New Fields       #
+    ###########################
+    _columns = {
+        'child_id': fields.many2one(
+            'compassion.child', _('Sponsored child'), readonly=True,
+            states={'draft': [('readonly', False)]}, ondelete='restrict'),
+        'child_name': fields.related(
+            'child_id', 'name', string=_('Sponsored child name'),
+            readonly=True, type='char'),
+        'child_code': fields.related(
+            'child_id', 'code', string=_('Sponsored child code'),
+            readonly=True, type='char'),
+        'partner_codega': fields.related(
+            'correspondant_id', 'ref', string=_('Partner ref'), readonly=True,
+            type='char'),
+        'correspondant_id': fields.many2one(

```

Comment:
You can move all correspondant functionality in module sponsorship_compassion. A regular contract will never have a correspondant.